satpy/tests/reader_tests/test_generic_image.py

```diff
@@ -180,10 +211,75 @@ def test_GenericImageFileHandler(self):
         self.assertTrue('transform' in dataset.attrs)
         self.assertTrue(np.all(np.isnan(dataset.data[:, :10, :10].compute())))
 
-        # Test masking of floats
+    def test_GenericImageFileHandler_masking_only_integer(self):
+        """Test direct use of the reader."""
+        from satpy.readers.generic_image import GenericImageFileHandler
+
+        class FakeGenericImageFileHandler(GenericImageFileHandler):
+
+            def __init__(self, filename, filename_info, filetype_info, file_content, **kwargs):
+                """Get fake file content from 'get_test_content'."""
+                if GenericImageFileHandler is object:
```

Comment:
When will this import ever return `object`? Also, does this class need to be inside the test function?

Reply:
It's basically a copy of an other fake reader (from file test_hdf5_utils.py). I supposed it is best practice t odo so (without knowning why..). Of cource I can remove it. I've already noted that some the checkers marked this as issue.
It is an inner class because all the other test routines in this file do their imports in the method itself. I thought, it would be better to do the same here too.

Reply:
Yeah I think now that we are using pytest for tests it isn't as important to have the various imports in the test functions themselves. @mraspaud thoughts?

Reply:
Just removed `if GenericImageFileHandler is object`.

Reply:
I tihnk it's fine without the check